querydsl-jpa/src/main/java/com/querydsl/jpa/hibernate/DefaultSessionHolder.java

```diff
@@ -33,12 +35,24 @@ public DefaultSessionHolder(Session session) {
 
     @Override
     public Query createQuery(String queryString) {
-        return session.createQuery(queryString);
+        try {
+            return MethodHack.createQuery.invoke(session, queryString);
```

Comment:
forgive my ignorance, but how does this solve the issue? Isn't there still the org.hibernate.Query vs org.hibernate.query.Query issue in the source code?

Reply:
As I understand it -- by using reflection to obtain and invoke the method, the exact method signature never gets compiled into the querydsl bytecode.  This allows it to call either hibernate 5.1 (and earlier) or 5.2 version of the method where the signature changed slightly from returning org.hibernate.Query to org.hibernate.query.Query.  Yes, the Query class moved, but they still have org.hibernate.Query in place (and deprecated) with the latter subclassing the former so some degree of polymorphism is still maintained and old source would still compile -- but once compiled it expects the compiled-in signature.